### DISCUSSION TITLE
UMAP cluster separation with bins vs. peaks

### DISCUSSION THREAD

**Discussion by AnjaliC4** [GreenleafLab/ArchR#1116] [Questions / Documentation] [answered]
Hi @rcorces, 

I compared 500 bp bins (estimated LSI: iterations=2, first selection= 'top', varfeatures=50,000, dims=1:30) and  Signac top features (q0 - all the features) after calling peaks with macs2. I noticed that even after tweaking several parameters such as increasing the iterations or decreasing the varfeatures, I get more separated clusters using peaks than bins. Here are the plots (first one bins and second peaks - in both cluster resolution=0.3). I really like the idea of using bins and understand that separation between UMAP clusters is more technical than biological, however, I am wondering if you can suggest which parameters to tweak to get more separated clusters?

In addition, I am having a bit of difficulty in understanding the use of the parameters provided in LSI. 
First, in selecting top features in the first iteration - how many top features are selected? Is there a way to change this number? 
Second, by tweaking which parameter can I perform LSI on all the cells instead of an estimated LSI. Basically, I think it has to be between these options - sampleCellsPre, projectCellsPre, and sampleCellsFinal- not sure which one exactly. 

Thank you very much!!

<img width="305" alt="Screen Shot 2021-10-28 at 1 27 43 PM" src="https://user-images.githubusercontent.com/60046859/139305342-0ccb29e6-18f5-44dc-8470-faf2b715ca04.png">

**Comment by rcorces** [comment, marked as answer]
To clarify, clusters are called in the LSI space, regardless of how they are plotted in UMAP space.
I think you would have better luck tweaking the parameters of the UMAP if you are looking to get more separation. for example `minDist`

You can also force `addIterativeLSI()` to mimic your results from Signac by tweaking the parameters. In our hands, using defined metrics for assessing clustering (i.e. not subjective measures like how separate the clusters are), ArchR performs very well.

**Reply by AnjaliC4** [reply]
Thank you Ryan, I totally agree that it is subjective to just say how separated the clusters are. I noticed that with Signac, I was using all the features (about 300,000 peaks) for LSI. Perhaps it is partly a matter of how many bins/peaks you provide for dimensionality reduction. I will now look into more defined metrics for assessing clusters. 

Could you please clarify my other questions -
First, In selecting top features in the first iteration - how many top features are selected? Is there a way to change this number?
Second, by tweaking which parameter can I perform LSI on all the cells instead of an estimated LSI. Is it projectCellsPre = TRUE? I am not sure if I am using it correctly.
Whenever I tweak any of these parameters - sampleCellsPre, projectCellsPre, and sampleCellsFinal, I get an error:
<simpleError in .logThis(sampledCellNames, "cellNames supplied", logFile = logFile): object 'logFile' not found>
I can report this issue separately if you'd prefer that.

Thank you very much.

**Reply by rcorces** [reply]
> First, in selecting top features in the first iteration - how many top features are selected? Is there a way to change this number?

`varFeatures` would specify this (see parameter definitions). However, things like `filterQuantile` also affect this so if you really want all features you would set `filterQuantile = 1`
<html>
<body>


varFeatures | The number of N variable features to use for LSI. The top N features will be used based on the selectionMethod.
-- | --



</body>
</html>

> Second, by tweaking which parameter can I perform LSI on all the cells instead of an estimated LSI. Basically, I think it has to be between these options - sampleCellsPre, projectCellsPre, and sampleCellsFinal- not sure which one exactly.

Correct. `projectCellsPre = FALSE` and setting both `sampleCellsPre` and `sampleCellsFinal` to `NULL` should do this.

**Comment by Brawni** [comment]
Hi @AnjaliC4! I have had similar observations with our data, I was wondering if you figured out how to get more separated clusters like with using Signac? 
Thank you!

**Reply by AnjaliC4** [reply]
Hi @Brawni , Since Signac uses peaks and ArchR uses bins for clustering - I believe the main differences come from this. 500bp bins are more likely to be similar across clusters than enriched peak regions. If you wish to use bins for clustering, there are several things you can try to get more separated clusters: 1. Use larger bin size (this will effect your gene activity and other downstream result). 2. If  you wish to use 500bp bins, then you can decrease the no. of varFeatures, reduce no. of dimsToUse, or change UMAP parameters (especially, play with spread and minDist). Evenafter, you will notice peaks tend to give slightly more separated clusters than bins - however, this will not mean that clusters are necessarily more distinct. Hope this helps!

**Reply by Brawni** [reply]
Thanks for your feedbacks! I played a bit with number of iterations, var features (decreasing them seem to give worse separation) but will give it a try with UMAP settings and bins. Do you know how to change bin size? Also, i did try to input PeakMatrix in `addIterativeLSI` but it generated an almost identical UMAP as with TileMatrix. Did you try this yourself? Ultimately, what strategy worked best for you? 
Thanks a lot!

**Reply by rcorces** [reply]
Thank you both for having this dialogue. Just adding a few things:
As @AnjaliC4 mentioned - ArchR uses bins and Signac uses peaks. Unless something has changed since the last time I looked at Signac, Signac also uses the peaks from the input peak x cell matrix which are peaks called on the bulk-ified sample. Its hard to imagine how this approach would be "better" at discriminating true cell types as it would be heavily biased towards peaks in the most prominent cell types. I think the analysis that we did in the paper comparing dimensionality reduction and clustering was quite fair and unbiased.

@Brawni - I'm not too surprised that TileMatrix and PeakMatrix give similar results since the interative LSI procedure selects a subset of bins, typically the most accessible elements for the first round.
